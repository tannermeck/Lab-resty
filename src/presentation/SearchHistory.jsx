import React from 'react';
import { Search } from './Search';

export const SearchHistory = () => {
    const item = localStorage.getItem('STORE');
    const array = JSON.parse(item);
    const object = array.map(obj => {
        return (
        <li key={obj.url}>
            <Search 
                verb={obj.method}
                link={obj.url}/>
        </li>
        )
    })
    return <ul>{object}</ul>
}
