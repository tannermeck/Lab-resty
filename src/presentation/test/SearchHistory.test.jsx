import React from 'react';
import { render } from '@testing-library/react';
import { SearchHistory } from '../SearchHistory.jsx';

describe('renders a list of methods and urls', () => {
    it('should render a list from local storage', () => {
        const fakeStorage = [{ method: 'GET', url: 'www.me.com' }]
        localStorage.setItem('STORE', JSON.stringify(fakeStorage))
        const { asFragment } = render(
            <SearchHistory />
        )
        expect(asFragment()).toMatchSnapshot();
    })
})