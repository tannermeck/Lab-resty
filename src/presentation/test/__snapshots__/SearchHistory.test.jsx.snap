// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of methods and urls should render a list from local storage 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <p>
        GET
      </p>
      <p>
        www.me.com
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
